Migrate camunda builtins from feel-editor to seperate repo (#1)
camunda/feel-builtins: README.md

# Camunda FEEL Builtins

Collection of builtin Camunda extensions for FEEL (Friendly Enough Expression Language). These builtins get extracted from [camunda-docs](https://github.com/camunda/camunda-docs/tree/main/docs/components/modeler/feel/builtin-functions).

## Installation

Install via npm:

```sh
npm install @camunda/feel-builtins
```

## Usage

> You can use the builtins in your FEEL editor by importing them from this package. Here's an example of how to integrate them into a FEEL editor instance:

Reviewer: Link `feel-editor`, consider integration with linting.

```suggestion
In your [FEEL editor](https://github.com/bpmn-io/feel-editor) you can use these builtins to establish the Camunda context:
```

Consider: Should this example also feature the Camunda dialect? I think that would make sense.

Consider: Should we also feature integration with feel-lint? I think that would make sense.

```js
import { camundaBuiltins } from '@camunda/feel-builtins';
import FeelEditor from '@bpmn-io/feel-editor';

const editor = new FeelEditor({
  container,
  builtins: camundaBuiltins
});
```

## Resources

* [Changelog](./CHANGELOG.md)
* [Issues](https://github.com/camunda/feel-builtins/issues)

> ## Setup and update builtins 

Reviewer: ```suggestion
## Build and Run
```

Prepare the project by installing dependencies:

```sh
npm install
```

Update the builtins
```sh
# pull camunda-docs and extract builtins
npm run update:builtins

> ```

Author: i removed testing due to this comment: https://github.com/camunda/feel-builtins/pull/1#discussion_r2205225939

> ```

Author: which runner should i use,?
I was thinking of just mocha (as in https://github.com/camunda/zeebe-bpmn-moddle/blob/main/package.json)

> ```

Reviewer 2: browser needed -> karma + mocha
no browser -> mocha

> ```

Author: added some basic tests

# Camunda FEEL Builtins

Collection of builtin Camunda extensions for FEEL (Friendly Enough Expression Language). These builtins get extracted from [camunda-docs](https://github.com/camunda/camunda-docs/tree/main/docs/components/modeler/feel/builtin-functions).

## Installation

Install via npm:

```sh
npm install @camunda/feel-builtins
```

## Usage

You can use the builtins in your FEEL editor by importing them from this package. Here's an example of how to integrate them into a FEEL editor instance:

```js
import { camundaBuiltins } from '@camunda/feel-builtins';
import FeelEditor from '@bpmn-io/feel-editor';

const editor = new FeelEditor({
  container,
  builtins: camundaBuiltins
});
```

## Resources

- [Changelog](./CHANGELOG.md)
- [Issues](https://github.com/camunda/feel-builtins/issues)

## Build and Run

Install dependencies:

```sh
npm install
```

Update camunda builtins
```sh
npm run builtins # pulls camunda-docs and extracts builtins
```

Build the project:

```sh

> npm run build

Author: I'm open to change, but I prefer individual `code` blocks per command. It makes it easier to only copy the command you actully want 
<img width="635" height="391" alt="image" src="https://github.com/user-attachments/assets/413b21de-48ed-4213-94cb-242452794c72" />

> npm run build

Reviewer: Let's be short and consistent ([diagram-js](https://github.com/bpmn-io/diagram-js), [bpmn-moddle](https://github.com/bpmn-io/bpmn-moddle), [bpmn-js](https://github.com/bpmn-io/bpmn-js), [Camunda Modeler](https://github.com/camunda/camunda-modeler), [bpmn-js-element-templates](https://github.com/bpmn-io/bpmn-js-element-templates)).

Developer getting started is just one part of the purpose of the README. The major part is usage, this is where we should aim for a great UX.

> npm run build

Author: as we no longer built i've only kept the install and update commands, which should be consistent with bpmn-js

# Camunda FEEL Builtins

Collection of builtin Camunda extensions for FEEL (Friendly Enough Expression Language). These builtins get extracted from [camunda-docs](https://github.com/camunda/camunda-docs/tree/main/docs/components/modeler/feel/builtin-functions).

## Installation

Install via npm:

```sh
npm install @camunda/feel-builtins
```

## Usage

### Feel Editor

In your [FEEL editor](https://github.com/bpmn-io/feel-editor) you can use these builtins to establish the Camunda context:

```js
import { camundaBuiltins } from '@camunda/feel-builtins';
import FeelEditor from '@bpmn-io/feel-editor';

const editor = new FeelEditor({
  container,
  builtins: camundaBuiltins,
  parserDialect: 'camunda'
});
```

### Feel Lint

With [@bpmn-io/feel-lint](https://github.com/bpmn-io/feel-lint) you can also use the builtins to lint expressions in the Camunda dialect:

```js
import { camundaBuiltins } from '@camunda/feel-builtins';
import { lintExpression } from '@bpmn-io/feel-lint';

function createContext(builtins) {

>   return builtins.reduce((context, builtin) => {

Reviewer: Yea, let's do that, and it is obvious from the README. A great benefit of documentation driven develoment.
